src/lib.rs

```diff
@@ -119,10 +119,10 @@ fn parse_sealed_impl(item_impl: syn::ItemImpl) -> syn::Result<TokenStream2> {
         sealed_path.push(parse_quote!(Sealed));
 
         let self_type = &item_impl.self_ty;
-        let (trait_generics, _, where_clauses) = &item_impl.generics.split_for_impl();
+        let trait_generics = &item_impl.generics.split_for_impl().1;
 
         Ok(quote! {
-            impl #trait_generics #sealed_path #arguments for #self_type #where_clauses {}
+            impl #trait_generics #sealed_path #arguments for #self_type {}
```

Comment:
I remember there was a reason you pointed out. But maybe I just misunderstood and took them out.
I'll put them back in, my bad!

Reply:
Yeah I think a distinction must be made between "necessary bounds to _ever_ `impl Sealed`" (the bounds on the trait _definition_), which, in that comment, I claimed they are not necessary (those will already be enforced by such bounds on the annotated trait) –with the exception of the `?Sized` "unbound", which is the only bound that actually _loosens_ genericity. So, technically, the `Sealed` _definition_ ought to carry `?Sized` unbounds on all its generic type parameters, and does not need anything more.

_Each_ `impl Sealed`, however, must have the same degree of generality than the annotated impl, so it has to carry / forward _all the bounds_ that the original impl had. This is less problematic than the trait definition, because we no longer need to be in a submodule and can thus just copy-paste, _verbatim_, such bounds without issues (whereas for the trait def, we cannot forward the clauses from without a submodule without a `use super::*;`, and such import breaks when it happens inside a function's body).

The situation is a bit subtle, granted but by thinking about this in advance we will be dodging many potential bugs / issues 🙂